packages/plugin-ext/src/hosted/node/hosted-plugin-localization-service.ts

```diff
@@ -62,26 +65,77 @@ export class HostedPluginLocalizationService implements BackendApplicationContri
             .then(() => this._ready.resolve());
     }
 
-    deployLocalizations(plugin: DeployedPlugin): void {
+    async deployLocalizations(plugin: DeployedPlugin): Promise<void> {
+        const disposable = new DisposableCollection();
         if (plugin.contributes?.localizations) {
             const localizations = buildLocalizations(plugin.contributes.localizations);
-            const versionedId = PluginIdentifiers.componentsToVersionedId(plugin.metadata.model);
-            this.localizationDisposeMap.set(versionedId, Disposable.create(() => {
+            disposable.push(Disposable.create(() => {
                 this.localizationProvider.removeLocalizations(...localizations);
-                this.localizationDisposeMap.delete(versionedId);
             }));
             this.localizationProvider.addLocalizations(...localizations);
         }
+        if (plugin.metadata.model.l10n || plugin.contributes?.localizations) {
+            disposable.push(await this.updateLanguagePackBundles(plugin));
+        }
+        if (!disposable.disposed) {
+            const versionedId = PluginIdentifiers.componentsToVersionedId(plugin.metadata.model);
```

Comment:
Just wondering: do we support multiple versions of the same plugin?

Reply:
I don't think so. I believe @colin-grant-work has a better grasp on that though.

Reply:
No, only one version can be active at once. At the moment, switching versions (e.g. using the `Install Another Version` command) requires a restart.